src/Bicep.Core/Semantics/FunctionOverloadBuilder.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Bicep.Core.Syntax;
+using Bicep.Core.TypeSystem;
+
+namespace Bicep.Core.Semantics
+{
+    public sealed class FunctionOverloadBuilder
+    {
+        private readonly string name;
+
+        private string? description;
+
+        private TypeSymbol returnType;
+
+        private readonly ImmutableArray<TypeSymbol>.Builder fixedParameterTypes;
+
+        private int minimumArgumentCount;
+
+        private int? maximumArgumentCount;
+
+        private TypeSymbol? variableParameterType;
+
+        private FunctionOverload.ReturnTypeBuilderDelegate returnTypeBuilder;
+
+        private FunctionFlags flags;
+
+        public FunctionOverloadBuilder(string name)
+        {
+            this.name = name;
+            this.returnType = LanguageConstants.Any;
+            this.fixedParameterTypes = ImmutableArray.CreateBuilder<TypeSymbol>();
+            this.returnTypeBuilder = args => LanguageConstants.Any;
+            this.variableParameterType = null;
+        }
+
+        public FunctionOverload Build() =>
+            new FunctionOverload(
+                this.name,
+                this.returnTypeBuilder,
+                this.returnType,
+                this.minimumArgumentCount,
+                this.maximumArgumentCount,
+                this.fixedParameterTypes.ToImmutable(),
+                this.variableParameterType,
+                this.flags);
+
+        public FunctionOverloadBuilder WithDescription(string description)
+        {
+            this.description = description;
+
+            return this;
+        }
```

Comment:
Doesn't look like `Build()` uses `description` at all - where does it get used?

Reply:
Not currently, but I will use it for generating the markdown in signature help.